torch/_functorch/_aot_autograd/jit_compile_runtime_wrappers.py

```diff
@@ -159,9 +243,13 @@ def aot_dispatch_autograd(
     aot_config: AOTConfig,
     *,
     fw_metadata: ViewAndMutationMeta,
-):
+) -> DispatchReturn:
+    flat_fn, flat_args, fw_metadata, pre_compile_wrappers = pre_compile(
+        flat_fn, flat_args, aot_config, needs_autograd=True, fw_metadata=fw_metadata
+    )
+
     fw_metadata.deterministic = torch.are_deterministic_algorithms_enabled()
```

Comment:
Is this call and the one to disable_amp below it too expensive to always call? It's only used in autograd, but I think the tradeoff of making it so fw_metadata is never mutated by any aot_dispatch_* function (outside of wrappers) is really nice.

These look like they go to C and just return a bool, so I can't imagine them being that expensive